### PR TITLE
#2313 fix bug in StateParamScheduler attach method

### LINKED ISSUE #2313: StateParamScheduler raises KeyError when save_history=true and multiple parameters are scheduled
## 🐛 Bug description
When scheduling multiple parameters on an engine, a `KeyError` exception is raised when `save_history=True`
To reproduce the bug: 
```python
    engine = Engine(lambda e, b: None)
    exp_state_parameter_scheduler1 = ExpStateScheduler(
        param_name="exp_scheduled_param1", initial_value=initial_value, gamma=gamma, save_history=True
    )
    exp_state_parameter_scheduler2 = ExpStateScheduler(
        param_name="exp_scheduled_param2", initial_value=initial_value, gamma=gamma, save_history=True
    )
    exp_state_parameter_scheduler1.attach(engine, Events.EPOCH_COMPLETED)
    exp_state_parameter_scheduler2.attach(engine, Events.EPOCH_COMPLETED)
    engine.run([0] * 8, max_epochs=max_epochs)
```
A fix should be applied here :
https://github.com/pytorch/ignite/blob/79678b916e1fbef702f861d71d5832e17c188351/ignite/handlers/state_param_scheduler.py#L55

```python
        if self.save_history:
            if not hasattr(engine.state, "param_history") or engine.state.param_history is None:  # type: ignore
                setattr(engine.state, "param_history", {})
            engine.state.param_history.setdefault(self.param_name, [])  # type: ignore[attr-defined]
```

## Environment

- PyTorch Version (e.g., 1.4):
- Ignite Version (e.g., 0.3.0):
- OS (e.g., Linux):
- How you installed Ignite (`conda`, `pip`, source):
- Python version:
- Any other relevant information:

### PR DESCRIPTION
**Pull request by fco-dv**
Fixes #2313

Description:
Fixing bug on StateParamScheduler attach method, add related test.
Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/handlers/test_state_param_scheduler.py [hunk 1]
```diff
@@ -155,6 +155,22 @@ def test_exponential_scheduler(max_epochs, initial_value, gamma):
     exp_state_parameter_scheduler.load_state_dict(state_dict)
 
 
+@pytest.mark.parametrize(
+    "max_epochs, initial_value, gamma", [(3, 10, 0.99)],
+)
```
Comment (sdesrozis) [inline tests/ignite/handlers/test_state_param_scheduler.py:160]: I don’t understand what is exactly tested. Maybe it is worth to add some assert. What do you think ?

**Comment by sdesrozis** [comment]
@fco-dv Thank you, I left a comment.

File: tests/ignite/handlers/test_state_param_scheduler.py (hunk 1 above)
Comment (fco-dv) [reply tests/ignite/handlers/test_state_param_scheduler.py:160]: Thanks for your comments @sdesrozis  , just wanted to test that no KeyError exception was raised, will push an update to the PR.

File: tests/ignite/handlers/test_state_param_scheduler.py [hunk 2]
```diff
@@ -155,6 +155,25 @@ def test_exponential_scheduler(max_epochs, initial_value, gamma):
...
+    engine = Engine(lambda e, b: None)
+    exp_state_parameter_scheduler1 = ExpStateScheduler(
+        param_name="exp_scheduled_param1", initial_value=initial_value, gamma=gamma, save_history=True
+    )
+    exp_state_parameter_scheduler2 = ExpStateScheduler(
+        param_name="exp_scheduled_param2", initial_value=initial_value, gamma=gamma, save_history=True
+    )
+    exp_state_parameter_scheduler1.attach(engine, Events.EPOCH_COMPLETED)
+    exp_state_parameter_scheduler2.attach(engine, Events.EPOCH_COMPLETED)
+    try:
```
Comment (KickItLikeShika) [inline tests/ignite/handlers/test_state_param_scheduler.py:171]: Can we find another way to do the test here?

**Review by KickItLikeShika — COMMENTED** [review]
Thanks for the PR @fco-dv, left a comment!

File: tests/ignite/handlers/test_state_param_scheduler.py (hunk 2 above)
Comment (fco-dv) [reply tests/ignite/handlers/test_state_param_scheduler.py:171]: @KickItLikeShika  thanks for your comment! , I am not fully satisfied either, will try to find another way..

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks @fco-dv

**Pull request merged**